### DISCUSSION TITLE
React and other frameworks

### DISCUSSION THREAD

**Discussion by devmnj** [lit/lit#3115] [Q&A] [answered]
Can I use lit with React,Vue,Svelte?

**Comment by devmnj** [comment]
ReferenceError: window is not defined   I got when I tried @svelte

**Reply by e111077** [reply]
The team is currently working on helper integration wrappers in other frameworks, but in the meantime [this article](https://css-tricks.com/using-custom-elements-in-svelte/) may help you with Svelte.

**Comment by drk-mtr** [comment]
Yes, it works with all of these frameworks. If you're having issues with Svelte I'd suggest sharing a minimal working repro on stackoverflow.

**Comment by devmnj** [comment]
I got it. 

I made a mistake

Thank you

**Reply by drk-mtr** [reply]
Yeah your code works fine for me :) I just needed to add imports - and I changed to not use self-closing tags as I feel like that's the norm.

```typescript
import "./styles.css";
import "./weather-element.js";
import "./sparkle-element.js";

export default function App() {
  return (
    <div className="App">
      <sparkle-element></sparkle-element>
      <weather-element></weather-element>
    </div>
  );
}
```

**Comment by devmnj** [comment]
I am facing problem in svelte which result in **window is not defined** 
 [the source component ](https://codesandbox.io/s/lit-react-component-kn7is8?file=/src/sparkle-element.js).


`<script>
 import '../lit/sparkle-element'
</script>
 <sparkle-element/>`

**Comment by e111077** [comment, marked as answer]
Web Components work in any framework that can support HTML. [Custom Elements Everywhere](https://custom-elements-everywhere.com/) should tell you about integration issues with other frameworks.

There may be some hoops you have to jump through in certain frameworks that don't expect web components in the server like NextJS, Nuxt, SvelteKit etc. The team is currently working on generating wrappers for frameworks such that it should be easier to integrate with their servers.

If you're just looking for helper wrappers you can see an initial implementation of the react wrapper in [@lit-labs/react](https://www.npmjs.com/package/@lit-labs/react). In the meantime stay tuned for helper wrappers for Angular and Vue as well as helper libraries for NextJS
